Name inference rule for masked select

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25569 Name inference rules for relu/relu_/threshold/threshold_
* #25568 Name inference rule for torch.cat
* #25567 Name inference for masked_fill_ / masked_fill
* **#25566 Name inference rule for masked select**

masked_select returns a tensor with None names. However, it broadcasts
its inputs so we need to perform a check that they are broadcastable.

Test Plan:
- new tests [namedtensor ci]

Differential Revision: [D17159071](https://our.internmc.facebook.com/intern/diff/D17159071)

zou3519 committed:
Name inference rule for masked select

masked_select returns a tensor with None names. However, it broadcasts
its inputs so we need to perform a check that they are broadcastable.

Test Plan:
- new tests [namedtensor ci]

nairbv commented on aten/src/ATen/native/LegacyDefinitions.cpp:
```diff
@@ -43,21 +44,30 @@ Tensor & masked_scatter__cpu(Tensor& self, const Tensor & mask, const Tensor & s
 Tensor masked_select_cpu(const Tensor & self, const Tensor & mask) {
+  Tensor result;
+#ifdef BUILD_NAMEDTENSOR
+  auto outnames = namedinference::compute_broadcast_outnames(self, mask);
```
`outnames` unused?

nairbv commented on aten/src/ATen/native/LegacyDefinitions.cpp:
```diff
@@ -43,21 +44,30 @@ Tensor & masked_scatter__cpu(Tensor& self, const Tensor & mask, const Tensor & s
 
 Tensor & masked_select_out_cpu(Tensor & result, const Tensor & self, const Tensor & mask) {
+#ifdef BUILD_NAMEDTENSOR
+  namedinference::compute_broadcast_outnames(self, mask);
```
this doesn't mutate anything right? is the call here just for error checking?

zou3519 replied:
The call is just here for error checking.

zou3519 committed:
Update on "Name inference rule for masked select"


masked_select returns a tensor with None names. However, it broadcasts
its inputs so we need to perform a check that they are broadcastable.

Test Plan:
- new tests [namedtensor ci]

Differential Revision: [D17159071](https://our.internmc.facebook.com/intern/diff/D17159071)
